hacker/models.py

```diff
@@ -14,92 +20,194 @@
     ('XXL', 'XXL'),
 ) 
 
-GENDER_CHOICES = (
+GENDERS = (
     ('M', 'Male'),
     ('F', 'Female'),
     ('NB', 'Non-binary'),
     ('NA', 'Prefer not to disclose'),
 )
 
-CLASSIFICATION_CHOICES = (
+CLASSIFICATIONS = (
     ('U1', 'U1'),
     ('U2', 'U2'),
     ('U3', 'U3'),
     ('U4', 'U4'),
     ('U5', 'U5'),
 )
 
-
-DIETARY_RESTRICTION_CHOICES = (
+DIETARY_RESTRICTIONS = (
     ('Vegan', 'Vegan'),
     ('Vegaterian', 'Vegarterian'),
     ('Halal', 'Halal'),
     ('Kosher', 'Kosher'),
     ('Food Allergies', 'Food Allergies'),
 )
 
-GRAD_YEAR_CHOICES = [(i,i) for i in range(timezone.now().year, timezone.now().year + 6)]        # TO-DO TEST
+WAVE_TYPES = (
+    ('Approve', 'Approve Application'),
+    ('Reject', 'Reject Application'),
+)
 
+GRAD_YEARS = [(i,i) for i in range(timezone.now().year, timezone.now().year + settings.EMAIL_CONFIRM_CODE_LENGTH)]
 
-class Hacker(AbstractUser):
-    admitted = models.NullBooleanField(blank=True)
-    checked_in = models.NullBooleanField(blank=True)
-    admitted_datetime = models.DateTimeField(null=True, blank=True)
-    checked_in_datetime = models.DateTimeField(null=True, blank=True)
 
-    # Overrides AbstractUser.first_name to require not blank
-    first_name = models.CharField(max_length=30, blank=False, verbose_name='first name')
+'''
+Misc. Information:
+
+    - 'first_name' overrides AbstractUser.first_name to require not blank
+    - 'last_name' overrides AbstractUser.last_name to require not blank
+    - 'email' overrides AbstractUser.email to require not blank
+
+'''
 
-    # Overrides AbstractUser.last_name to require not blank
+class Hacker(AbstractUser):
+    first_name = models.CharField(max_length=30, blank=False, verbose_name='first name')
     last_name = models.CharField(max_length=150, blank=False, verbose_name='last name')
-    
-    # Overrides AbstractUser.email to require not blank
     email = models.EmailField(blank=False)
 
+    checked_in = models.NullBooleanField(blank=True)
+    email_confirmed = models.BooleanField(blank=True, default=False)
+
+    checked_in_datetime = models.DateTimeField(null=True, blank=True)
+
+    confirm_code = models.CharField(max_length=6, blank=True, null=True)
+
+    def has_related_application(self):
+        a = getattr(self, 'application', None)
+        return a is not None
+
+    def has_related_confirmation(self):
+        c = getattr(self, 'confirmation', None)
+        return c is not None
+
+    def has_related_team(self):
```

Comment:
In this case, I don't think we should be checking for a `confirmation` as well as a `team`, nor should we package this as a function.